### DISCUSSION TITLE
Adding DSL functions to to join field?

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#409] [Ideas]
Hi,
several times before running a join, I need to transform a field. In example I need to capitalize it.

It would be great to have something like the code below and be able to use mlr DSL functions.

```
mlr --csv join --ul -j fieldOne -l fieldOne -r 'uppercase(fieldTwo)' ...
```

What about it?

**Comment by aborruso** [comment]
@johnkerl ping :)

**Comment by johnkerl** [comment]
Hi @aborruso -- sorry for the delay.

My recommendation would be a 'decorate, join, undecorate' which also has the fancy name of [Schwartzian transform](https://en.wikipedia.org/wiki/Schwartzian_transform) -- a fancy way of adding: add a column using `put`; do the join; remove it using `cut -x`.

I think this would suffice ... and I hope so, since `put`/`filter` are already complex in their DSL, and `join` is already complex in its join logic ... if we can do this by leveraging each verb's strengths rather than adding one's complexities to the other, that would be nice ...

**Reply by aborruso** [reply]
Ok you are right.

Once again I have learned something

**Reply by trantor** [reply]
Ah, the good old Schwartzian transform... as a perler I much appreciate the reference :smile:
